lib/timers.js

```diff
@@ -316,6 +319,11 @@ exports.setTimeout = function(callback, after) {
 
   after *= 1; // coalesce to number or NaN
 
+  if (after > TIMEOUT_MAX) {
```

Comment:
`NaN` is like `0` or `-1`, and also coerces to `1`, right? If so, that's probably not surprising.

I'm going off the fact that the max timeout might be surprising.